### ISSUE TITLE
Document how checksum is generated, and provide guides for how to recreate

### ISSUE THREAD

**Issue by perolavsvendsen** [equinor/fmu-dataio#487]
Data consumers should be able to validate that the data they get after it has passed through multiple pipelines is identical to what `fmu-dataio` exported in the first place.

Documentation should contain a description of how the checksum has been made, and how data consumers can re-generate and compare in their end to validate data integrity.

**Comment by janbjorge** [comment]
This would be easy if we have http like structure going. As in a header (meta about the meta) and a payload. In the header we could have version, hash etc... Without, the problem becomes recursive (to create the hash, you need the data, but to have the data you need hash).

Top of my head example.

```yaml
- header:
  hash: ...
  version: ...
- payload:
  <existing fmu-dataio>
```

**Comment by perolavsvendsen** [comment]
They have the data + metadata. They want to validate that the data they have is in fact the correct data.

**Comment by janbjorge** [comment]
We have that already? Under _file_ there a md5checksum?

**Comment by perolavsvendsen** [comment]
I'm not saying we cannot change it going forward, but how we currently generate the hash should be in the docs as a reference.

**Comment by perolavsvendsen** [comment]
> We have that already? Under _file_ there a md5checksum?

Yes, that's the one 👍

**Comment by janbjorge** [comment]
> > We have that already? Under _file_ there a md5checksum?
> 
> Yes, that's the one 👍

Sorry, think my dyslexia got me there. When _they_ try to md5 on their end, are they getting different results?

**Comment by perolavsvendsen** [comment]
Hopefully not... I probably didn't provide enough context here.

We are qualifying the use of fmu-dataio + Sumo + SSDL to automate data flow from FMU workflows to SSDL. One question asked by the approvers were "how do we know that REP is getting the right data" (aka "how do we know that somewhere in the various pipelines, data didn't get mixed up").

The assumed fix is to document how consumers of FMU results can validate (using the checksum). I think this belongs in the fmu-dataio docs, as it would be valid independent of Sumo. And it belongs with the documentation of the rest of the data model. 

So, a recipe for how to confirm data validity, essentially.

**Comment by janbjorge** [comment]
I should be straight forward, without any ambiguity(assuming they are using a well know standard libery) to compute the md5 sum for file. Did a quick test locally, and the builtin md5 in Mac gives the same as the one in fmu-dataio.

```bash
(dataio) ➜  fmu-dataio git:(main) ✗ md5 share/results/maps/mysurface.gri
MD5 (share/results/maps/mysurface.gri) = eb6a16d365586da79ea54e0f4dc172d2
(dataio) ➜  fmu-dataio git:(main) ✗ python3 -c"from fmu.dataio._utils import md5sum;print(md5sum('share/results/maps/mysurface.gri'))"
eb6a16d365586da79ea54e0f4dc172d2
```

**Comment by perolavsvendsen** [comment]
Yes, I don't see why it should be optional to make it. Only thing I can think of is when no file is being generated, e.g in a cloud service where the data go directly from instance to Sumo.

If I remember correctly, fmu-dataio creates the file on disk, then creates the checksum.

Perhaps we need data.checksum instead.

Also, this won't actually work with e.g tables or seismic, where the data is fragmented when stored.

In short, this may need more thinking... 🤔

**Comment by janbjorge** [comment]
Ive added a few _todos_ that came to think if right now. I think even if to goes directly to sumo a hash would be useful for many purpeases. One case could be: But is there a chance that we get the exact same data and in that case there is no need for re-upload?

**Comment by perolavsvendsen** [comment]
Possibly 🤷‍♂️

Nevertheless the whole things is starting to feel like a topic we should refine a bit.
